### ISSUE TITLE
Is it possible to "press" two keys at once with ADDKEY or AUTOTYPE?

### ISSUE THREAD

**Issue by Banjo-Oz** [joncampbell123/dosbox-x#2159]
I was looking for a keyboard stuffer when I discovered DOSBox-X already includes the utilities ADDKEY and AUTOTYPE.

Is there a way to press two keys at once (or hold down one key then press the other) using those programs?

For example, to send "F11+f" (toggle fullscreen) via ADDKEY or AUTOTYPE as part of a batch file.

I can't seem to figure out how to do it, if it is possible.

**Comment by Wengier** [comment]
I think this is certainly an interesting question. I don't really think it is possible to press two keys at once with either command at this time, but I guess allowing holding down one key then pressing another key is a good suggestion. Alternatively, perhaps we can allow these commands to activate mapper functions, such as the toggle fullscreen function. Good suggestion!

**Comment by Banjo-Oz** [comment]
Thanks for looking at this and considering it! I actually had tried SCANCODE, KEY-FAKE and a couple of other old DOS-era keystuffers first and couldn't get it to work, then I discovered the built-in DOSBox-X tools (ADDKEY and AUTOTYPE) but as I said, couldn't get those to work either.

To be honest, the F11+f combo is a bit wonky on my machine anyway... I need to be fairly "deliberate" when I press and hold F11, then press f. It feels a bit more sensitive than the usual ALT+ENTER for some reason.

Regardless, just having a "FULLSCREEN" and "WINDOW" command would be really, really helpful.

I can set my other config settings right from a batch file (e.g. aspect from false to true, fullscreen to true) but actually making the window move to fullscreen can't be done without F11+f right now, it seems.

**Comment by rderooy** [comment]
There is a config command for this, but it has no effect once dosbox-x is started.
```
config -set fullscreen=true
```

**Comment by Wengier** [comment]
@Banjo-Oz With updated code in #2171 you can now turn on full-screen mode with a command like ```config -set fullscreen=true``` and ```config -set fullscreen=false```. Hopefully this will do exactly what you want.

By the way, the pixel-perfect scaling as you desired is already working. You probably want to check out #2167 for more information.

**Comment by Banjo-Oz** [comment]
Wow, that's awesome news! So setting it via config will actually toggle my window to fullscreen (and back) and not just set the config setting? Previously, it didn't work (set the setting, but needed a reboot or F11+f). Is this in a build I can test for you?

**Comment by Wengier** [comment]
@Banjo-Oz Yes, with the new code the command will actually toggle the full screen mode. You can try the following Windows binary for the feature:

* [dosbox-x-update.zip](http://condos.fam.cx/dosware/dosbox-x-update.zip)

**Comment by Banjo-Oz** [comment]
@Wengier Just FYI: using that build works for setting fullscreen or windowed, but when I use "config-set sdl fullscreen=false" once the change happens it seems like there are also multiple "enter" keypresses entered too (the prompt repeats multiple times, hence why I suspect the system is "pressing enter" about a dozen times). Not sure if it's just on my machine or not?

**Comment by Wengier** [comment]
@Banjo-Oz There may be some keyboard-related issues during the full-screen toggle. I will look at it further.

**Comment by Wengier** [comment]
@Banjo-Oz The following Windows build will hopefully fix the said issue:

* [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)

Please check it out.
